inference-engine/src/plugin_api/cpp_interfaces/interface/ie_iplugin_internal.hpp

```diff
@@ -166,6 +166,21 @@ class IInferencePlugin : public std::enable_shared_from_this<IInferencePlugin> {
     virtual std::shared_ptr<IExecutableNetworkInternal> LoadNetwork(const CNNNetwork& network,
                                                                     const std::map<std::string, std::string>& config,
                                                                     RemoteContext::Ptr context) = 0;
+
+    /**
+     * @brief Creates an executable network from model file path
+     * @param modelPath A path to model
+     * @param config A string-string map of config parameters relevant only for this load operation
+     * @return Created Executable Network object
+     */
+    virtual ExecutableNetwork LoadNetwork(const std::string& modelPath,
+                                          const std::map<std::string, std::string>& config)
+    {
+        auto cnnNet = GetCore()->ReadNetwork(modelPath, std::string());
+        auto net = GetCore()->LoadNetwork(cnnNet, GetName(), config);
```

Comment:
@ilya-lavrenov GetCore()->LoadNetwork returns ExecutableNetwork, that's why this method returns `ExecutableNetwork`, not `shared_ptr<IExecutableNetworkInternal>` . Is it ok? It looks a bit unusual, but I didn't find another way without significant changes

Reply:
Good question, in general I don't mind.

CC @apankratovantonp do you have any ideas about such consistency / inconsistency?

Reply:
If you do not create `ExecutableNetwork` manually and just use object that was created by `Core` everything is OK. But if you create `ExecutableNetwork` by yourself you have to preserve SO objects that was used to load that network. 
Certainly internal API should return some internal entities not public C++ API. But this refactoring is not implemented yet. So you can leave everything as is.